pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -49,6 +49,10 @@ func ValidateMachinePool(p *openstack.MachinePool, ci *CloudInfo, controlPlane b
 			logrus.Warnf("Volume size is recommended to be greater than %d GB to use root volumes, had %d GB", recommendedStorage, p.RootVolume.Size)
 		}
 
+		if len(p.Zones) > 0 && len(p.RootVolume.Zones) == 0 {
```

Comment:
just a note for the reviewers, there is no CEL validation in the installer (e.g. `XValidation:rule`) which could have been handy too.